### ISSUE TITLE
Concerto CLI fails on some Accord Project models

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/concerto#89]
**Describe the bug**
The CLI (e.g., `concerto generate`) fails on well formed and valid models from the Accord Project.

**To Reproduce**

E.g., using examples from [cicero-models](https://github.com/accordproject/models/):
```
bash-3.2$ cd ~/git/cicero-models/src/
bash-3.2$ concerto --version
0.80.4-20191003155034
bash-3.2$ concerto generate --ctoFiles signature/signature.cto 
Class ContractSigned is not declared as abstract. It must define an identifying field. File 'signature/signature.cto': line 9 column 1, to line 11 column 2.  IllegalModelException: Class ContractSigned is not declared as abstract. It must define an identifying field. File 'signature/signature.cto': line 9 column 1, to line 11 column 2. 
bash-3.2$ concerto generate --ctoFiles cicero/contract.cto 
Generated JSONSchema code.
bash-3.2$ concerto generate --ctoFiles cicero/runtime.cto 
Class Request is not declared as abstract. It must define an identifying field. File 'cicero/runtime.cto': line 13 column 1, to line 13 column 23.  IllegalModelException: Class Request is not declared as abstract. It must define an identifying field. File 'cicero/runtime.cto': line 13 column 1, to line 13 column 23. 
bash-3.2$ 
```

**Additional context**
This is most likely due to distinct systems models. See also #62

**Comment by vipul-vinsol** [comment]
I wanna work on this one if no one else is working on this @jeromesimeon

**Comment by vipulbhj** [comment]
I wanna work on this, can I try this if no one else is working on it @jeromesimeon

**Comment by dselman** [comment]
The issue is that the `ModelManager` is created without a system model:
https://github.com/accordproject/concerto/blob/master/packages/concerto-cli/lib/commands.js#L64

In AP we use this system model:
https://github.com/accordproject/ergo/blob/master/packages/ergo-compiler/lib/apmodelmanager.js#L21

A quick fix would be to use the same system model for `concerto generate` but we really should remove system models altogether to ensure that models are completely self-describing and reusable.

**Comment by vipulbhj** [comment]
If that is the case, I should patch this first with the above suggested approach and then once the bug is fixed we can decide how to remove system model completely ??

**Comment by jeromesimeon** [comment]
@vipulbhj Sounds like a pretty good plan.

For the first step, one question will be _which_ system model since Accord Project has one and at least Hyperledger Composer has another, and other users may have used that capability and built their own. One possibility would be to have an option `--ctoSystem` which lets the user specify a system model. If we have a default one, which one should it be. I would vote for the Accord Project one which is here: https://github.com/accordproject/models/blob/master/src/cicero/base.cto

The second step needs definitely a more completely proposal since this will affect every other project, and users. We need to decide what each kind of base type mean (`Asset`, `Concept`, `Event`, etc). I'll be working on this, so I'm happy to join forces!

**Comment by jeromesimeon** [comment]
For reference, the way to add the system model to the model manager is illustrated here (using the Accord Project system model): https://github.com/accordproject/ergo/blob/9e216d86c87c6b67f640e7982f0e639a792e6207/packages/ergo-compiler/lib/apmodelmanager.js#L44

**Comment by vipulbhj** [comment]
Thanks for the reference material. For the first step this does sound like a good plan. I be trying to dig more in the command code today and I will try to add the option as discussed. I am relatively new here so I am gone go with your suggestion for default, if that's okay

**Comment by jeromesimeon** [comment]
That's great @vipulbhj ! If you start a PR with some of the basic idea in place for that new option, I can review while the discussion about default model takes place.

Since `cicero-cli` is brand new, we do not have to worry whether this breaks anyone's code.

**Comment by jeromesimeon** [comment]
Just a caveat: this will probably a short lived fix, since we'll likely remove the notion of system model soon after!

**Comment by vipulbhj** [comment]
I will try to submit a PR soon

**Comment by vipulbhj** [comment]
Hey @jeromesimeon, make some first rough draft changes. Mind taking a look ??

**Comment by jeromesimeon** [comment]
Fixed in #113 . Thanks for your help @vipulbhj !
